docs/GDPs/index.rst

```diff
@@ -0,0 +1,59 @@
+=====================================
+GT4Py Development Proposals
+=====================================
+
+This page provides an overview of the development decisions and priorities
+for GT4Py. Specifically, it contains a collections of GT4Py Development
+Proposals (GDPs) with proposed features for the framework in various
+stages of discussion or completion (see `GDP 0 <gdp-0000>`__).
```

Comment:
I think this only works properly in the rendered output, not in GitHub preview. I would leave it as it is unless it doesn't work properly when we clone the GDPs folder to the project wiki

Reply:
![Screenshot from 2020-03-25 16-22-16](https://user-images.githubusercontent.com/5988437/77553287-e8f8a780-6eb4-11ea-800a-f9f3c087ad32.png)

Reply:
this is literally looking at your branch and not the preview

Reply:
That's what I mean: you need to run sphinx on the rst to generate the proper HTML with the working links, but it doesn't work out of the box in the automatic preview rendered by github.

Reply:
Ah I see.

I think we should make an effort to have it looking good on github too since the discussion will be held here, right?

Reply:
I've tried to fix the format issues but I couldn't find a way to make the references between templates work as intended for Sphinx and the preview at Github at the same time. I still think it is more important to make the format work for Sphinx, because this is the tool used to generate the actual rendered  HTML/PDF/whatever documents, so I vote to leave it as it is.

I also mentioned inthe PR description that the final idea is to publish GDPs  with the proper format in the project wiki once the discussion is finished. This can be done easily with Pandoc and fully automatized with Github actions, so users/readers could check the wiki to read the GDPs instead of previewing the .rst files in the repo.